src/app/users/users.service.ts

```diff
@@ -190,9 +193,30 @@ export class UsersService {
         // Make copy of user so UI doesn't change until DB change succeeds
         const tempUser = { ...user, roles: newRoles };
         observers.push(this.couchService.put('_users/org.couchdb.user:' + tempUser.name, tempUser));
+        this.sendNotifications(user);
       }
       return observers;
     }, [])).pipe(map((responses) => this.stateService.requestData(this.dbName, 'local')));
   }
 
+  sendNotifications(user) {
+    const link = '/manager/users';
+    const notificationDoc = {
+      user: user._id,
+      'message': 'You were assigned a new role',
+      link,
+      linkParams: { activeTab: 'taskTab' },
+      'type': 'newRole',
+      'priority': 1,
+      'status': 'unread',
+      'time': this.couchService.datePlaceholder,
+      userPlanetCode: user.userPlanetCode
+    };
+    return this.couchService.findAll(
+      'notifications',
```

Comment:
This needs a specific query since this should be checking to see if a role change notification already exists for this user.  Right now this `_find` query is only checking to see if any notifications exist for the entire system.